test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/DnsHostNameIsResolvableAttribute.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Net;
+using Microsoft.AspNetCore.Testing.xunit;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.FunctionalTests
+{
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
+    public class DnsHostNameIsResolvableAttribute : Attribute, ITestCondition
+    {
+        private string _hostname;
+
+        public bool IsMet
+        {
+            get
+            {
+                try
+                {
+                    _hostname = Dns.GetHostName();
+                    var addresses = Dns.GetHostAddressesAsync(_hostname).GetAwaiter().GetResult();
+                    if (addresses.Any(i => !IPAddress.IsLoopback(i)))
```

Comment:
The behavior doesn't change. We try to bind to IPv6Any, but request made to localhost/hostname should still succeed.